ablySpec/RestChannel.swift

```diff
@@ -198,14 +222,228 @@ class RestChannel: QuickSpec {
+
+                    it("as a JSON Array") {
+                        client.httpExecutor = mockExecutor
+                        // JSON Array
+                        waitUntil(timeout: testTimeout) { done in
+                            channel.publish(array, callback: { error in
+                                expect(error).to(beNil())
+
+                                if let request = mockExecutor.requests.last, let http = request.HTTPBody {
+                                    // Array
+                                    let json = JSON(data: http)
+                                    expect(json["data"].asArray).to(equal(array))
+                                    expect(json["encoding"].string).to(equal("json"))
+                                }
+                                else {
+                                    XCTFail("No request or HTTP body found")
+                                }
+                                done()
+                            })
+                        }
+                    }
+
+                    it("as a JSON Object") {
+                        client.httpExecutor = mockExecutor
+                        // JSON Object
+                        waitUntil(timeout: testTimeout) { done in
+                            channel.publish(dictionary, callback: { error in
+                                expect(error).to(beNil())
+
+                                if let request = mockExecutor.requests.last, let http = request.HTTPBody {
+                                    // Dictionary
+                                    let json = JSON(data: http)
+                                    expect(json["data"].asDictionary).to(equal(dictionary))
+                                    expect(json["encoding"].string).to(equal("json"))
+                                }
+                                else {
+                                    XCTFail("No request or HTTP body found")
+                                }
+                                done()
+                            })
+                        }
+                    }
+
+                }
+
+                // RSL4d4
+                it("messages received should be decoded based on the encoding field") {
+                    let cases = [text, integer, decimal, dictionary, array, binaryData]
+
+                    cases.forEach { caseTest in
+                        waitUntil(timeout: testTimeout) { done in
+                            channel.publish(caseTest, callback: { error in
+                                expect(error).to(beNil())
+                                done()
+                            })
+                        }
+                    }
+
+                    var totalReceived = 0
+                    try! channel.history(nil) { result, error in
+                        expect(error).to(beNil())
+                        expect(result).toNot(beNil())
+                        expect(result?.hasNext).to(beFalse())
```

Comment:
I'm changing it as I review the tests, but because this is not yet on master it would be helpful if you do it before merging.